engine/moduleManager/modules/rendering/RenderingModule.cpp

```diff
@@ -152,22 +159,34 @@ SDL_Texture* RenderingModule::LoadTexture(const SpriteRenderable& sprite, int& w
 	return result.second ? result.first->second.get() : nullptr;
 }
 
-void RenderingModule::RenderTexture(SDL_Texture* texture, const ComponentTransform& transform) {
+void RenderingModule::RenderTexture(SDL_Texture* texture, const ComponentTransform& transform, const glm::vec4* sourceRect) {
 	if (!texture) {
 		std::cerr << "Cannot render null texture." << std::endl;
 		return;
 	}
 
 	SDL_FRect destRect = {
-			(transform.position.x - transform.scale.x / 2.0f),
-			(transform.position.y - transform.scale.y / 2.0f),
+			transform.position.x - transform.scale.x / 2.0f,
+			transform.position.y - transform.scale.y / 2.0f,
 			transform.scale.x,
 			transform.scale.y
 	};
 
+	SDL_Rect sdlSourceRect;
+	if (sourceRect) {
+		sdlSourceRect = {
+				static_cast<int>(sourceRect->x),
```

Comment:
sourceRect kan denk ik gwn in AnimationRenderable glm::ivec4 worden, dan is het überhaupt al integers en hoeft het niet gecast te worden